rotkehlchen/chain/evm/tokens.py

```diff
@@ -97,9 +101,34 @@ def generate_multicall_chunks(
 
 
 class EvmTokens():
-    def __init__(self, database: DBHandler, manager: EthereumManager):
+    def __init__(
+            self,
+            database: DBHandler,
+            manager: EthereumManager,
+            makerdao_vaults_module: Optional['MakerdaoVaults'],
+    ):
         self.db = database
         self.manager = manager
+        self.makerdao_vaults_module = makerdao_vaults_module
+
+    def _proxies_for_addresses(
+        self,
+        addresses: List[ChecksumEvmAddress],
+    ) -> Tuple[List[ChecksumEvmAddress], Dict[ChecksumEvmAddress, ChecksumEvmAddress]]:
+        """
+        Accepts a list of address to find proxies for. Returns a list of original addresses with
+        added proxies and a mapping of proxy addresses to owner addresses.
+        """
+        proxies_mapping = {}
+        if self.makerdao_vaults_module is not None:
+            proxies_mapping = self.makerdao_vaults_module._get_accounts_having_proxy()
+            proxies_mapping = {v: k for k, v in proxies_mapping.items()}
+            for proxy_address, owner_address in proxies_mapping.items():
+                if owner_address not in addresses:
+                    continue
```

Comment:
can't you add this condition to the `proxies_mapping` dict comprehension? Once that is done I believe you can just concatenate

```
addresses = addresses + list(proxies_mappings.keys())
```

unless you need all the addresses for some reason in the return statmenet